src/docs/tutorials/04-edit-email-templates-with-flex-cli/index.md

```diff
@@ -0,0 +1,240 @@
+   :template :booking-request-accepted}
+  ,,,
+  {:name :notification/review-by-customer-second,
+   :on :transition/review-2-by-customer,
+   :to :actor.role/provider,
+   :template :review-by-other-party-published}]
+```
+
+A template for a notification is a directory that is named after the
+`:template` value and contains two files:
+
+- `TEMPLATE_NAME-subject.txt` - holds the mail Subject line template
+- `TEMPLATE_NAME-html.html` - contains the template for the HTML version of the mail
+
+Both parts are mandatory. All emails that are sent from the
+marketplace contain both the HTML and plain text variants and the
+recipient's mail client is free to choose which one to visualize and
+how. The text version is automatically generated from the HTML
+template.
+
+### Example
+
+For example, the `:notification/new-booking-request` notifiction:
+
+```clojure
+{:name :notification/new-booking-request,
+ :on :transition/confirm-payment,
+ :to :actor.role/provider,
+ :template :new-booking-request}
+ ```
+
+has the following template:
+
+```
+new-booking-request/
+├── new-booking-request-html.html
+└── new-booking-request-subject.txt
+```
+
+Note that the template name (e.g. `:new-booking-request`) doesn't have
+to match the notification name
+(e.g. `:notification/new-booking-request`) as you can share a template
+with multiple notifications.
+
+## Email template syntax
+
+Templates use [Handlebars](http://handlebarsjs.com/) syntax.
+
+Example HTML:
+
+```handlebars
+{{~#*inline "format-date"~}}
+{{date date format="MMM d, YYYY"}}
+{{~/inline~}}
+
+<html>
+  <body>
+    {{#with transaction}}
+    <h1>Please respond to a request by {{customer.display-name}}</h1>
+
+    <p>
+      Good news! {{customer.display-name}} just requested to book {{listing.title}}
+      from {{> format-date date=booking.start}} to {{> format-date date=booking.end}}.
+      Here's the breakdown.
+    </p>
+
+    ...
+
+    {{/with}}
+
+    <hr />
+
+    <p>
+      You have received this email notification because you are a member of {{marketplace.name}}.
+      If you no longer wish to receive these emails, please contact {{marketplace.name}} team.
+    </p>
+  </body>
+</html>
+
+```
+
+Variables within `{{ }}` are expanded and escaped, so that they are
+safe to place inside HTML content. As seen above, some variables have
+nested values, which can be accessed with dot `.` operator.
+
+The template syntax supports conditionals, loops, helpers and other
+constructs. For details see the [Email
+templates](/references/email-templates/) reference.
+
+## Change a template
+
+Let's change the email template for new booking requests.
+
+Open the `new-booking-request/new-booking-request-html.html` file in a
+text editor and make a simple change:
+
+```diff
+- <h1>Please respond to a request by {{customer.display-name}}</h1>
++ <h1>Please respond to a new booking request by {{customer.display-name}}</h1>
+```
+
+When you've made the change, save the file.
```

Comment:
What do you think, would it make sense to add a short example on how to access the attributes of the context with Handlebars? In a way that for somebody who has no prior experience on templating languages it would become clear how to traverse the context with `{{#with transaction}}` and `{{customer.display-name}}` for example, and how those map to the context description.

Reply:
I was already thinking if I should mention those in the tutorial, so maybe. On the other hand, I think one of the most important things in tutorials is IMO this:

> **Don’t explain anything the learner doesn’t need to know in order to complete the tutorial.** Extended discussion is important - just not in a tutorial. In a tutorial, it is an obstruction and a distraction. Only the bare minimum is appropriate. Instead, link to explanations elsewhere in the documentation.

from https://www.divio.com/blog/documentation/

I think the goal of this tutorial is to make **a** change in a template and to show the workflow with Flex CLI, and there are already quite a few steps to achieve that. Could the example be another article or do you think it could be short enough to add to this one?

Reply:
@kpuputti good point. I was thinking one or two code examples how to access the context variables. Or, maybe it could be enough to highlight from the _Example HTML_ that how is it exactly selecting the `transaction.customer.display-name` attribute.

Reply:
I'd say our epic budget can now leave this change out to get things forward. Is that ok for you? The email template improvement epic can improve the docs as well then.

Reply:
@kpuputti sounds good and was about to suggest the same.